sklearn/metrics/cluster/_supervised.py

```diff
@@ -150,12 +151,18 @@ def contingency_matrix(
            [0, 1, 1],
            [1, 0, 1]])
     """
-
     if eps is not None and sparse:
         raise ValueError("Cannot set 'eps' when sparse=True")
 
-    classes, class_idx = np.unique(labels_true, return_inverse=True)
-    clusters, cluster_idx = np.unique(labels_pred, return_inverse=True)
+    xp, is_array_api_compliant, device = get_namespace_and_device(
+        labels_pred, labels_true
+    )
+
+    if sparse and is_array_api_compliant:
+        raise ValueError("Cannot use sparse=True while using array api dispatch")
```

Comment:
Coming back to this, maybe raising this exception is unnecessarily annoying. We could instead just return

Accepting array API inputs would not necessarily mean to output a data structure of the same type when output is request explicitly to use another container type.

Reply:
> We could instead just return

Did you mean to write something after `return`?

Reply:
Yes, I mean to return a sparse scipy data-structures (backed by numpy arrays).